packages/components/MenuButton/src/MenuButton.tsx

```diff
@@ -156,4 +124,30 @@ export const MenuButton = compose<MenuButtonType>({
   },
 });
 
+const SubMenuItem: React.FunctionComponent<MenuButtonItemProps> = (props: MenuButtonItemProps): JSX.Element => {
+  const [showSubmenuState, setShowSubmenu] = React.useState(false);
+  const toggleShowSubmenu = React.useCallback(() => {
+    setShowSubmenu(!showSubmenuState);
+  }, [showSubmenuState, setShowSubmenu]);
+  const onDismissSubmenu = React.useCallback(() => {
+    setShowSubmenu(false);
+  }, [setShowSubmenu]);
+
+  const { showSubmenu = showSubmenuState, submenuProps, componentRef, submenuItems, onHoverIn = toggleShowSubmenu, ...restItems } = props;
+  const { onDismiss = onDismissSubmenu, setShowMenu = toggleShowSubmenu, ...restSubmenuProps } = submenuProps;
+
+  return (
+    <React.Fragment>
+      <SubmenuItem componentRef={componentRef} onHoverIn={onHoverIn} {...restItems} />
+      {showSubmenu && (
+        <Submenu target={componentRef} onDismiss={onDismiss} setShowMenu={setShowMenu} {...restSubmenuProps}>
+          {submenuItems?.map((submenuItem) => (
```

Comment:
It shouldn't unless something has changed - the "key" prop should make it so that React will "map" the generated items to what already exist and only replace the elements that have been changed.

Reply:
https://reactjs.org/docs/lists-and-keys.html#keys